commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java

```diff
@@ -232,6 +334,33 @@ public boolean hasManageCommercePaymentMethodGroupRelsPermission() {
 			CommerceActionKeys.MANAGE_COMMERCE_PAYMENT_METHODS);
 	}
 
+	public boolean isLastRecurringPaymentMethod(String engineKey) {
```

Comment:
why?

Reply:
If its the last recurring payment method we disable subscription products since there's no longer a payment method to purchase them with.